core/components/atoms/avatar/avatar.tsx

```diff
@@ -137,14 +142,15 @@ class Avatar extends React.Component<IAvatarProps, IAvatarState> {
     const { imageErrored, gravatarErrored } = this.state
     const { email, initials, icon, image } = this.props
 
+    // NOTE: The following order of checks MATTER. They determine the fallback
+    //       precedence when there are any missing props or failed attempts.
+    //       Carefully review all scenarios are covered when changing the
+    //       following lines.
```

Comment:
maybe adding a unit test for this function can prevent this from breaking the logic in the future